hide-ignored/hide-ignored-flags.js

```diff
@@ -1,31 +1,66 @@
+// @version      1.0
+// @description  Hides those posts that you've already seen in the flag dashboard
 // @author       bhargavrao
+// @author       Glorfindel
+// @match        *://*.stackexchange.com/admin/*
 // @match        *://*.stackoverflow.com/admin/*
+// @match        *://*.stackapps.com/admin/*
+// @match        *://*.askubuntu.com/admin/*
+// @match        *://*.superuser.com/admin/*
+// @match        *://*.serverfault.com/admin/*
+// @match        *://*.mathoverflow.net/admin/*
+// @include      *://*.stackexchange.com/admin/*
 // @include      *://*.stackoverflow.com/admin/*
+// @include      *://*.stackapps.com/admin/*
+// @include      *://*.askubuntu.com/admin/*
+// @include      *://*.superuser.com/admin/*
+// @include      *://*.serverfault.com/admin/*
+// @include      *://*.mathoverflow.net/admin/*
 // @grant        none
 // ==/UserScript==
 
 (function() {
-    $('.s-sidebarwidget:first').append('<div class="s-sidebarwidget--content sm:pt8 sm:pb8"><fieldset><div class="grid gs8 gsx"><div class="grid--cell"><input class="s-checkbox" id="hide-ignored" type="checkbox"></div><div class="grid--cell"><div class="s-label fw-normal"><span>Hide Ignored Posts</span><p class="js-toggle-apply-filters s-description sm:d-none">Hide posts that you have already visited</p></div></div></div></fieldset></div>')
-    $('#hide-ignored').change(
-            function(){
-                if ($(this).is(':checked')) {
-                    $(".visited-post").each(
-                        function(index){
-                            $(this).hide();
-                        }
-                    );
-                }
-                if ($(this).is(':checkbox:not(:checked)')) {
-                    $(".visited-post").each(
-                        function(index){
-                            $(this).show();
-                        }
-                    );
-                }
-            });
-})();
+  // Event handler
+  let onChange = function(newValue) {
+    // Remember value
+    if (newValue) {
+      window.localStorage.setItem("HideTheVisited", true);
+    } else {
+      window.localStorage.removeItem("HideTheVisited");
+    }
+    // Show/hide posts
+    $(".visited-post").each(function(index) {
+      if (newValue) {
+        $(this).hide();
+      } else {
+        $(this).show();
+      }
+    });
+  };
+  
+  // Add checkbox and label
+  let grid = $("input.js-toggle-apply-filters").parent().parent().parent();
+  grid.append(`
+                                <div class="grid gs8 gsx">
+                                    <div class="grid--cell">
+                                        <input class="s-checkbox hide-visited" type="checkbox" id="hide-visited">
+                                    </div>
+                                    <div class="grid--cell">
+                                        <div class="s-label fw-normal">
+                                            <label for="hide-visited">Hide visited posts</label>
```

Comment:
Can you add the description back? I had this earlier, but it can be anything:

```<p class="js-toggle-apply-filters s-description sm:d-none">Hide posts that you have already visited</p>```